app/controllers/SessionsCategoryController.scala

```diff
@@ -0,0 +1,274 @@
+          val subCategoryExists = subCategoryList.contains(cleanSubCategory.toLowerCase)
+
+          if (subCategoryExists) {
+            Future.successful(BadRequest("Sub-category already exists"))
+          } else {
+            sessionsRepository
+              .updateSubCategoryOnChange(oldSubCategoryName, newSubCategoryName)
+              .flatMap { session =>
+                if (session.ok) {
+                  categoriesRepository
+                    .modifySubCategory(categoryName, oldSubCategoryName, cleanSubCategory)
+                    .map {
+                      result =>
+                        if (result.ok) {
+                          Logger.info(s"Sub-category was successfully modified $newSubCategoryName")
+                          Ok("Successfully Modified sub category")
+                        } else {
+                          Logger.error(s"Something went wrong while modifying sub-category $newSubCategoryName")
+                          BadRequest("Got an error while modifying sub category")
+                        }
+                    }
+                } else {
+                  Future.successful(BadRequest("Got an error on updating session table"))
+                }
+              }
+          }
+        }
+    }
+  }
+
+  def deletePrimaryCategory(categoryId: String): Action[AnyContent] = superUserAction.async { implicit request =>
+    if (categoryId.isEmpty) {
+      Future.successful(BadRequest("Please select a valid primary category"))
+    } else {
+      categoriesRepository
+        .getCategories
+        .flatMap { categories =>
+          val category = categories.find(_._id.toString() == categoryId)
+
+          category.fold {
+            Future.successful(BadRequest("No such primary category exists"))
+          } { category =>
+            if (category.subCategory.isEmpty) {
+              sessionsRepository
+                .updateCategoryOnChange(category.categoryName, "")
+                .flatMap { session =>
+                  if (session.ok) {
+                    categoriesRepository
+                      .deletePrimaryCategory(category._id.stringify)
+                      .map { result =>
+                        if (result.ok) {
+                          Logger.info(s"Primary category with categoryId $categoryId was successfully deleted")
+                          Ok("Primary category was successfully deleted")
+                        } else {
+                          Logger.error(s"Something went wrong while deleting primary category with category Id $categoryId")
+                          BadRequest("Got an error while deleting")
+                        }
+                      }
+                  } else {
+                    Future.successful(BadRequest("Got an error in session table"))
+                  }
+                }
+            } else {
+              Future.successful(BadRequest("First delete all its sub category"))
```

Comment:
Replace "First delete all its sub category" with "All sub categories should be deleted prior to deleting the primary category"